tutoroars/plugin.py

```diff
@@ -263,7 +293,8 @@ def load_xapi_test_data(num_batches: int, batch_size: int) -> list[tuple[str, st
             "echo 'Making demo xapi script executable...' && "
             "chmod +x /app/oars/scripts/clickhouse-demo-xapi-data.sh && "
```

Comment:
nit: 

```suggestion
            "chmod +x /app/oars/scripts/oars/clickhouse-demo-xapi-data.sh && "
```

Reply:
Hmm.. even with the above nit fixed, this init command isn't working. I'm getting :

```
(tutor) $ tutor dev do load-xapi-test-data
<snip>
Starting tutor_dev_redis-permissions_1 ... done
Creating tutor_dev_oars-job_run        ... done
<snip>
clickhouse_connect.driver.exceptions.OperationalError: Error HTTPConnectionPool(host='clickhouse', port=8123): Max retries exceeded with url: /?wait_end_of_query=1 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fb25f4b4370>: Failed to establish a new connection: [Errno 111] Connection refused')) executing HTTP request http://clickhouse:8123
```

I also noticed from the output above that the oars job container doesn't ensure that clickhouse is running, but it is:

```
(tutor) $ docker ps | grep clickhouse
20:4a481dab7285   clickhouse/clickhouse-server:23.3                "/entrypoint.sh"         2 hours ago   Up 33 minutes               0.0.0.0:8123->8123/tcp, :::8123->8123/tcp, 0.0.0.0:9000->9000/tcp, :::9000->9000/tcp, 9009/tcp   tutor_dev_clickhouse_1
```

Reply:
Ah, I'm also getting connection errors to clickhouse when I try to init OARS:

```
(tutor) $ tutor dev do init -l oars
Initialising all services...
Running init task in clickhouse
docker-compose -f /data/jill/.tutor/env/local/docker-compose.yml -f /data/jill/.tutor/env/dev/docker-compose.yml -f /data/jill/.tutor/env/dev/docker-compose.tmp.yml --project-name tutor_dev -f /data/jill/.tutor/env/local/docker-compose.jobs.yml -f /data/jill/.tutor/env/dev/docker-compose.jobs.yml -f /data/jill/.tutor/env/dev/docker-compose.jobs.tmp.yml run --rm clickhouse-job sh -e -c 'echo "Initialising Clickhouse..."
ch_connection_max_attempts=10
ch_connection_attempt=0
until clickhouse client --user ch_admin --password="XYTMmE6LVqYT6AcO8mdvizvB" --host "clickhouse"  --port 9000 -q '"'"'exit'"'"'
do
    ch_connection_attempt=$(expr $ch_connection_attempt + 1)
    echo "    [$ch_connection_attempt/$ch_connection_max_attempts] Waiting for Clickhouse service (this may take a while)..."
    if [ $ch_connection_attempt -eq $ch_connection_max_attempts ]
    then
      echo "Clickhouse initialisation error" 1>&2
      exit 1
    fi
    sleep 10
done
echo "Clickhouse is up and running"'
Creating tutor_dev_clickhouse-job_run ... done
Initialising Clickhouse...
Code: 210. DB::NetException: Connection refused (clickhouse:9000). (NETWORK_ERROR)

    [1/10] Waiting for Clickhouse service (this may take a while)...
Code: 210. DB::NetException: Connection refused (clickhouse:9000). (NETWORK_ERROR)

    [2/10] Waiting for Clickhouse service (this may take a while)...
<snip>
```

So there seems to be something broken with the docker containers connecting to clickhouse?

Reply:
@bmtcril I rebuilt my tutor dev stack and everything looked like it came up ok: the superset container could `curl http://clickhouse:8123`, and the `ch_report` user existed with the expected password..

But the OpenedX Clickhouse database wouldn't connect. The sqlalchemy url `clickhousedb+connect://ch_report:XXXXXXXXXX@clickhouse:8123/xapi` was throwing this error:

```
superset_1                   | 2023-06-14 02:14:43,676:WARNING:superset.views.base:SupersetErrorsException
superset_1                   | Traceback (most recent call last):
superset_1                   |   File "/app/superset/databases/commands/test_connection.py", line 120, in run
superset_1                   |     raise DBAPIError(None, None, None)
superset_1                   | sqlalchemy.exc.DBAPIError: (builtins.NoneType) None
superset_1                   | (Background on this error at: http://sqlalche.me/e/13/dbapi)
superset_1                   |
superset_1                   | The above exception was the direct cause of the following exception:
superset_1                   |
superset_1                   | Traceback (most recent call last):
<snip>
superset_1                   | superset.databases.commands.exceptions.DatabaseTestConnectionFailedError: [SupersetError(message='(builtins.NoneType) None\n(Background on this error at: http://sqlalche.me/e/13/dbapi)', error_type=<SupersetErrorType.GENERIC_DB_ENGINE_ERROR: 'GENERIC_DB_ENGINE_ERROR'>, level=<ErrorLevel.ERROR: 'error'>, extra={'engine_name': 'ClickHouse Connect', 'issue_codes': [{'code': 1002, 'message': 'Issue 1002 - The database returned an unexpected error.'}]})]
```

So I looked up the connection string in [Superset's docs](https://superset.apache.org/docs/databases/installing-database-drivers/#supported-databases-and-dependencies) and they say to use `clickhousedb://` instead of `clickhousedb-connect://` as the scheme.

And sure enough, changing it to `clickhousedb://ch_report:XXXXXXXXXX@clickhouse:8123/xapi` worked!

But it gets weirder: if I manually add a Clickhouse Connect database using the Superset wizard:

![clickhouse_connect_db](https://github.com/openedx/tutor-contrib-oars/assets/7556571/cc272867-832c-4201-a7cc-8c8ad2ec2c17)

and then export it, it exports with `sqlalchemy_uri: clickhousedb+connect://ch_report:XXXXXXXXXX@clickhouse:8123/xapi`! So I think this change might be in transition.. Not sure what exactly we should do at this moment though.